include/support.h

```diff
@@ -27,12 +27,19 @@
 #include <cstdio>
 #include <cstring>
 #include <limits>
+#include <memory>
 #include <stdexcept>
 #include <string>
 #include <vector>
 
 #include <SDL.h>
 
+// General-use unique-pointer types
+void sdl_thread_deleter(SDL_Thread *t);
+using sdl_thread_ptr_t = std::unique_ptr<SDL_Thread, decltype(&sdl_thread_deleter)>;
+using sdl_mutex_ptr_t = std::unique_ptr<SDL_mutex, decltype(&SDL_DestroyMutex)>;
+using sdl_cond_ptr_t = std::unique_ptr<SDL_cond, decltype(&SDL_DestroyCond)>;
+
```

Comment:
Please don't put these type redefinitions project-wide.

In some contexts `sdl_thread_deleter` will not be appropriate. In other context we will prefer raw pointers or other types of pointers besides unique_ptr.

If these need to be used, it's best to limit them in scope to the single class or file that uses them.

Reply:
Thanks, @dreamer.  

Scoping these (as well as `mt32_service_t` that lived outside the class) made for a very nice clean up.  I also opted for less annotated type names, which I think improves readability.

Fixed in:
https://github.com/dosbox-staging/dosbox-staging/pull/786/commits/983febeee54960b5e564df2cd2d9b913e8da6005